assignments/models.py

```diff
@@ -0,0 +1,149 @@
+    status = models.CharField(_('status'), max_length=128, choices=STATUS_CHOICES)
+    budget = models.IntegerField(_('budget'), default=0)
+    slug = models.CharField(max_length=128, unique=True)
+
+    class Meta:
+        verbose_name = _('assignment')
+        verbose_name_plural = _('assignments')
+        unique_together = ('name', 'area')
+
+    def save(self, *args, **kwargs):
+        if self.slug is None:
+            self.slug = slugify('{}{}'.format(self.name, self.area)[:100])
+        return super(Assignment, self).save(*args, **kwargs)
+
+    def __str__(self):
+        return self.name
+
+
+class Student(models.Model):
+    """
+    Learning environment user
+    """
+    SEX_CHOICES = (
+        ('male', _('Male')),
+        ('female', _('Female'))
+    )
+    school = models.CharField(_('school name'), max_length=256)
+    school_class = models.CharField(_('class'), max_length=256)
+    sex = models.CharField(_('sex'), choices=SEX_CHOICES, max_length=128)
+
+    def __str__(self):
+        return '{} ({})'.format(self.name, self.school_class)
+
+
+class BaseTask(models.Model):
+    """
+    Common fields for all tasks
+    """
+    title = models.CharField(_('title'), max_length=256)
+    description = models.TextField(_('description'))
+
+    class Meta:
+        abstract = True
+
+
+class Task(BaseTask):
+    """
+    Assignment is divided in tasks.
+    Task is defined by title, description, and can contain smaller tasks of various types.
+    """
+    environment = models.ForeignKey(Assignment)
+
+    class Meta:
+        verbose_name = _('task')
+        verbose_name_plural = _('tasks')
+
+
+class OpenTextTask(BaseTask):
+    """
+    Simple question task where the answer is placed in text area
+    """
+    task = models.ForeignKey(Task, related_name='open_text_tasks')
+
+    class Meta:
+        verbose_name = _('open text task')
+        verbose_name_plural = _('open text tasks')
+
+    def __str__(self):
+        return 'Open text task: {}'.format(self.title)
+
+
+class AnswerOpenTextTask(models.Model):
+    """
+    Answer to related OpenTextTask.
+    """
+    student = models.OneToOneField(Student)
+    answer = models.CharField(_('answer'), max_length=256)
+    open_text_task = models.OneToOneField(OpenTextTask)
+
+
+class MultiChoiceTask(BaseTask):
+    """
+    Question tasks with multi choice answer
+    """
+    task = models.ForeignKey(Task, related_name='multi_choice_tasks')
+
+    class Meta:
+        verbose_name = _('multi choice task')
+        verbose_name_plural = _('multi choice tasks')
+
+    def __str__(self):
+        return 'Multi choice task: {}'.format(self.title)
+
+    @property
+    def right_answers(self):
+        return self.multichoice_choices.filter(right_answer=True)
+
+
+class Choice(models.Model):
+    """
+    Choice for related multi choice task
+    """
+    text = models.CharField(_('text'), max_length=128)
+    right_answer = models.BooleanField(_('right answer'))
```

Comment:
We don't need `right_answer` feature yet so it is better to plan as a whole in later if it needed.